Add more REST API docs (assets, campaigns, categories, fields) (#100)
mautic/developer-documentation-new: docs/rest_api/categories.rst

Categories
##########

> Use this endpoint to obtain details on Mautic'sCategories or to manipulate Category memberships.

Reviewer: ```suggestion
Use this endpoint to obtain details on Mautic's Categories, or to manipulate Category memberships.
```

**Using Mautic's API Library**

You can interact with this API through the :xref:`Mautic API Library` as follows, or use the various http endpoints as described in this document.

.. code-block:: php

   <?php
   use Mautic\MauticApi;
   use Mautic\Auth\ApiAuth;

   // ...
   $initAuth    = new ApiAuth();
   $auth        = $initAuth->newAuth($settings);

>    $apiUrl      = "https://your-mautic.com";

Reviewer: ```suggestion
   $apiUrl      = "https://example.com";
```

   $api         = new MauticApi();
   $categoryApi = $api->newApi("categories", $auth, $apiUrl);

.. vale off

Get Category
************

.. vale on

.. code-block:: php

   <?php

   //...
   $category = $categoryApi->get($id);

.. code-block:: json

   {  
     "category":{  
       "id":221,
       "title":"test",
       "alias":"test4",
       "description":null,
       "color":null,
       "bundle":"asset"
     }
   }

Get an individual Category by ID.

.. vale off

**HTTP Request**

.. vale on

``GET /categories/ID``

**Response**

``Expected Response Code: 200``

See JSON code example.

> **Category Properties**

Reviewer: ```suggestion
**Category properties**
```

.. list-table::
   :header-rows: 1

   * - Name
     - Type
     - Description
   * - ``id``
     - int
     - ID of the Category
   * - ``isPublished``
     - boolean
     - Published status of the Category
   * - ``dateAdded``
     - ``datetime``

>      - Date/time Category got created

Reviewer: ```suggestion
     - Category creation date/time
```

   * - ``createdBy``
     - int
     - ID of the User that created the Category
   * - ``createdByUser``
     - string
     - Name of the User that created the Category
   * - ``dateModified``
     - datetime/null
     - Date/time Category was last modified
   * - ``modifiedBy``
     - int
     - ID of the User that last modified the Category
   * - ``modifiedByUser``
     - string
     - Name of the User that last modified the Category
   * - ``title``
     - string
     - The Category title
   * - ``alias``
     - string
     - The Category alias
   * - ``description``
     - string
     - The Category description
   * - ``color``
     - string
     - The Category color
   * - ``bundle``
     - string
     - The bundle where the Category is available

.. vale off

> List contact categories

Reviewer: ```suggestion
List Contact Categories
```

***********************

.. vale on

.. code-block:: php

   <?php

   //...
   $categories = $categoryApi->getList($searchFilter, $start, $limit, $orderBy, $orderByDir, $publishedOnly, $minimal);

> Returns a list of ContactCategories available to the User. This list isn't filterable.

Reviewer: ```suggestion
Returns a list of Contact Categories available to the User. This list isn't filterable.
```
I think this is correct but please check for technical accuracy.

         "description":null,
         "color":"b36262",
         "bundle":"point"
       },
     ]
   }

**Category Properties**

.. list-table::
   :header-rows: 1

   * - Name
     - Type
     - Description
   * - ``id``
     - int
     - ID of the Category
   * - ``isPublished``
     - boolean
     - Published status of the Category
   * - ``dateAdded``
     - ``datetime``
     - Date/time Category got created
   * - ``createdBy``
     - int
     - ID of the User that created the Category
   * - ``createdByUser``
     - string
     - Name of the User that created the Category
   * - ``dateModified``
     - datetime/null
     - Date/time Category was last modified
   * - ``modifiedBy``
     - int
     - ID of the User that last modified the Category
   * - ``modifiedByUser``
     - string
     - Name of the User that last modified the Category
   * - ``title``
     - string
     - The Category title
   * - ``alias``
     - string
     - The Category alias
   * - ``description``
     - string
     - The Category description
   * - ``color``
     - string
     - The Category color
   * - ``bundle``
     - string
     - The bundle where the Category is available

.. vale off

> Create category

Reviewer: ```suggestion
Create Category
```
